src/modules/Popup/Popup.js

```diff
@@ -0,0 +1,270 @@
+
+    /** A Flowing popup have no maximum width and continue to flow to fit its content */
+    flowing: PropTypes.bool,
+
+    /** Takes up the entire width of its offset container */
+    fluid: PropTypes.bool,
+
+    /** Header displayed above the content in bold */
+    header: PropTypes.string,
+
+    /** Whether the popup should not close on hover */
+    hoverable: PropTypes.bool,
+
+    /** Invert the colors of the popup */
+    inverted: PropTypes.bool,
+
+    /** The node where the popup should mount.. */
+    hideOnScroll: PropTypes.bool,
+
+    /** Horizontal offset in pixels to be applied to the popup */
+    offset: PropTypes.number,
+
+    /** Event triggering the popup */
+    on: PropTypes.oneOf(['hover', 'click', 'focus']),
+
+    /** Positionning for the popover */
+    position: PropTypes.string,
+
+    /** Popup size */
+    size: PropTypes.string,
+
+    /** custom popup style */
+    style: PropTypes.object,
+
+    /** Element to be rendered in-place where the popup is defined. */
+    trigger: PropTypes.node,
+
+    /** Popup width */
+    width: PropTypes.string,
+  }
+
+  static defaultProps = {
+    position: 'top left',
+    on: 'hover',
+  }
+
+  static _meta = _meta
+  static Content = PopupContent
+  static Header = PopupHeader
+
+  state = {}
+
+  computePopupStyle(positions) {
+    const style = { position: 'absolute' }
+    const { clientWidth, clientHeight } = document.documentElement
+    const { pageXOffset, pageYOffset } = window
+    const { offset } = this.props
+
+    if (_.includes(positions, 'right')) {
+      style.right = Math.round(clientWidth - (this.coords.right + pageXOffset))
+      style.left = 'auto'
+    } else if (_.includes(positions, 'left')) {
+      style.left = Math.round(this.coords.left + pageXOffset)
+      style.right = 'auto'
+    } else {  // if not left nor right, we are horizontally centering the element
+      const xOffset = (this.coords.width - this.popupCoords.width) / 2
+      style.left = Math.round(this.coords.left + xOffset + pageXOffset)
+      style.right = 'auto'
+    }
+
+    if (_.includes(positions, 'top')) {
+      style.bottom = Math.round((clientHeight - pageYOffset) - this.coords.top)
+      style.top = 'auto'
+    } else if (_.includes(positions, 'bottom')) {
+      style.top = Math.round(this.coords.bottom + pageYOffset)
+      style.bottom = 'auto'
+    } else {  // if not top nor bottom, we are vertically centering the element
+      const yOffset = (this.coords.height + this.popupCoords.height) / 2
+      style.top = Math.round(this.coords.bottom + pageYOffset - yOffset)
+      style.bottom = 'auto'
+
+      const xOffset = this.popupCoords.width + 8
+      if (_.includes(positions, 'right')) {
+        style.right -= xOffset
+      } else {
+        style.left -= xOffset
+      }
+    }
+
+    if (offset) {
+      if (_.isNumber(style.right)) {
+        style.right -= offset
+      } else {
+        style.left += offset
+      }
```

Comment:
I thought of the same thing, but it seems like SUI offset is only acting on the X axis and from left to right. I can change to one of your suggestion if you want me to